crates/bevy_reflect/src/serde/de/deserializer.rs

```diff
@@ -14,6 +14,193 @@ use crate::{
 use core::{fmt, fmt::Formatter};
 use serde::de::{DeserializeSeed, Error, IgnoredAny, MapAccess, Visitor};
 
+/// Allows overriding the default deserialization behaviour of
```

Comment:
Now that #8611 is merged, the docs should probably mention (at least briefly) `DeserializeWithRegistry` and how this differs than that, and when you might want to prefer one over the other.

Reply:
Added
```rs
/// # Compared to [`DeserializeWithRegistry`]
///
/// [`DeserializeWithRegistry`] allows you to define how your type will be
/// deserialized by a [`TypedReflectDeserializer`], given the extra context of
/// the [`TypeRegistry`]. If your type can be deserialized entirely from that,
/// then you should prefer implementing that trait instead of using a processor.
///
/// However, you may need more context-dependent data which is only present in
/// the scope where you create the [`TypedReflectDeserializer`]. For example, in
/// an asset loader, the `&mut LoadContext` you get is only valid from within
/// the `load` function. This is where a processor is useful, as the processor
/// can capture local variables.
```